dist2src/core.py

```diff
@@ -139,15 +139,46 @@ def create_tag(self, tag, branch):
         """Create a Git TAG at the tip of BRANCH"""
         self.repo.create_tag(tag, ref=branch, force=True)
 
-    def cherry_pick_base(self, from_branch, to_branch):
+    def get_tags_for_head(self) -> List[str]:
+        return list(
+            tag.name for tag in self.repo.tags if tag.commit == self.repo.head.commit
+        )
+
+    def cherry_pick_base(self, from_branch, to_branch, theirs=False):
         """Cherry-pick the first commit of a branch
 
         Cherry-pick the first commit of FROM_BRANCH to TO_BRANCH in the
         repository stored in GITDIR.
         """
         num_commits = sum(1 for _ in self.repo.iter_commits(from_branch))
         self.checkout(to_branch, create_branch=True)
-        self.repo.git.cherry_pick(f"{from_branch}~{num_commits - 1}")
+        git_options = (
+            {"strategy_option": "theirs", "keep_redundant_commits": True}
+            if theirs
+            else {}
+        )
+        self.repo.git.cherry_pick(f"{from_branch}~{num_commits - 1}", **git_options)
+
+    def revert_to_ref(self, ref, message=None):
+        message = message or f"Revert the state to {ref}"
+        # https://git-scm.com/book/en/v2/Git-Tools-Reset-Demystified
+        # Reset index without changing HEAD
+        self.repo.git.reset(ref, ".")
+        self.commit(message)
+        # clear the working-tree
+        self.repo.git.reset("HEAD", hard=True)
+
+    def clean(self):
+        """
+        Clean the repo.
+        """
+        # We need to use two `force` options
+        # to remove the submodules/git repos as well.
+        self.repo.git.clean(x=True, d=True, f=True, force=True)
```

Comment:
How about: 

```suggestion
        self.repo.git.clean("-xdff")
```

This would make the double-force less confusing.

Reply:
Good idea. I hope it will work since `xdff=True` did not work.

Reply:
Works well. I'll leave the comment there to document that behaviour.